File: src/net/opgenorth/yeg/buildings/data/IBuildingDataService.java
package net.opgenorth.yeg.buildings.data;

import net.opgenorth.yeg.buildings.model.Building;
import net.opgenorth.yeg.buildings.model.IBuildingSorter;

import java.util.List;

public interface IBuildingDataService {
    List<Building> fetchAll();
    List<Building> fetchAll(IBuildingSorter sortedBy);
    boolean hasRecords();
}

File: dotnet/yegbuildings/IBuildingDataService.cs
using System.Collections.Generic;
using net.opgenorth.yeg.buildings.model;

namespace net.opgenorth.yeg.buildings
{
    public interface IBuildingDataService
    {
        IList<Building> FetchAll();
        IList<Building> FetchAll(IBuildingSorter sortedBy);
        bool HasRecords { get;  }
    }

    public interface IBuildingSorter
    {
        IList<Building> Sort(IList<Building> buildings);
    }
}